src/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+add_subdirectory(vmec_interface)
+
+
+add_executable(stella.x  ${SOURCES} )
+
+#set flags here
+#TODO add the most common flags needed for stella
+
+set(dialect "-ffree-line-length-none -fno-backslash -I /usr/include")
+set(bounds "-fbounds-check")
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${dialect} -march=native")
+set(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} ${bounds} -pedantic -g -Wall -fimplicit-none")
+if(PRECISION STREQUAL "DOUBLE")
+	set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8 -fdefault-double-8")
+endif()
+
+#get package includes and libs here
+
+find_package(MPI REQUIRED)
+if(MPI_FOUND)
+	message(STATUS "Found MPI Libraries in ${MPI_LIBRARIES}")
+	get_filename_component (MPI_LIBRARY_PATH ${MPI_LIBRARY} DIRECTORY)
+    target_include_directories(stella.x PRIVATE ${MPI_Fortran_INCLUDE_PATH})
+	target_link_libraries( stella.x PRIVATE ${MPI_Fortran_LIBRARIES})
+endif()
+
+find_package(LAPACK REQUIRED)
+if(LAPACK_FOUND)
+	message(STATUS "Found LAPACK Libraries in ${LAPACK_LIBRARIES}")
+	target_link_libraries( stella.x PRIVATE ${LAPACK_LIBRARIES})
+endif()
+
+
+find_package(FFTW REQUIRED)
+if(FFTW_FOUND)
+	message(STATUS "Found FFTW Libraries in ${FFTW_LIBRARIES}")
+	include_directories(${FFTW_INCLUDES})
+	target_link_libraries(stella.x PRIVATE ${FFTW_LIBRARIES})
```

Comment:
Exactly which FFTW libraries to link depends on whether stella is built with double or single precision -- I've got a `FindFFTW.cmake` module that finds them both, and then we can automatically link the correct one here.